### PR TITLE
Use gomega.Equal instead of reflect.DeepEqual

### PR DESCRIPTION
**Pull request by VirrageS**
**What this PR does / why we need it**: This MR changes usage of `reflect.DeepEqual` to standard `gomega.Equal` function to simplify the code.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: None

**Special notes for your reviewer**: None

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/migration_test.go [hunk 1]
```diff
@@ -3699,11 +3698,11 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred(), "should have been able to retrive the VMI instance")
 			cpuSetTarget := getVirtLauncherCPUSet(vmi)
-			Expect(reflect.DeepEqual(cpuSetSource, cpuSetTarget)).To(BeFalse(), "source CPUSet %v should be != than target CPUSet %v", cpuSetSource, cpuSetTarget)
+			Expect(cpuSetSource).NotTo(Equal(cpuSetTarget), "source CPUSet %v should be != than target CPUSet %v", cpuSetSource, cpuSetTarget)
```
Comment (xpivarc) [inline tests/migration_test.go:3701]: Do we require the print now? `"source CPUSet %v should be != than target CPUSet %v", cpuSetSource, cpuSetTarget)` ?

**Review by xpivarc — COMMENTED** [review]
/approve

File: tests/migration_test.go (hunk 1 above)
Comment (VirrageS) [reply tests/migration_test.go:3701]: I'm not sure how the `gomega` will print this by default, but you are right, this seems unnecessary.

I'll remove that.

Comment (xpivarc) [reply tests/migration_test.go]: This is what you get 
```
  Expected
      <[]int | len:2, cap:2>: [1, 2]
  not to equal
      <[]int | len:2, cap:2>: [1, 2]
```

So I would just leave the comment(`CPUSet of source launcher should not match targets one`).

Comment (VirrageS) [reply tests/migration_test.go]: Done :)

**Review by maiqueb — CHANGES_REQUESTED** [review]
Just noticed this. Probably that's why the [unit tests lane is not passing](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7712/pull-kubevirt-unit-test/1524373929794211840).

**Comment by xpivarc** [comment]
/hold 
Test failures are unrelated. An investigation is ongoing.

**Pull request merged**